store/sqlstore/session_store.go

```diff
@@ -52,7 +52,12 @@ func (me SqlSessionStore) Save(session *model.Session) (*model.Session, *model.A
 
 	session.PreSave()
 
-	tcs := me.Team().GetTeamsForUser(session.UserId)
+	tcs := make(chan store.StoreResult, 1)
+	go func() {
+		teams, err := me.Team().GetTeamsForUser(session.UserId)
+		tcs <- store.StoreResult{Data: teams, Err: err}
+		close(tcs)
```

Comment:
There's a chance for `tcs` channel to be closed before `rtcs := <-tcs` ? Am I missing something?
@rvillablanca @jespino

Reply:
No, this is a buffered channel and there won't be any problem if we try to read after close. See the section "Reads and Writes to Closed Channels" here https://www.rapidloop.com/blog/golang-channels-tips-tricks.html to understand it better.

Reply:
Thanks for the explanation @jespino